drivers/hmis/app/models/hmis/ce/match/rule.rb

```diff
@@ -17,18 +17,31 @@ def eligibility_requirement?
       rule_type == ELIGIBILITY_REQUIREMENT
     end
 
+    scope :eligibility_requirement, -> do
+      where(rule_type: ELIGIBILITY_REQUIREMENT)
+    end
+
     def priority_scheme?
       rule_type == PRIORITY_SCHEME
     end
 
+    scope :priority_scheme, -> do
+      where(rule_type: PRIORITY_SCHEME)
+    end
+
     def applies_to_opportunity?(opportunity)
-      config = applicability_config || {}
+      config = applicability_config.symbolize_keys || {}
       applicability = OpportunityApplicability.new(
         owner: owner,
         project_types: config[:project_types],
         project_funders: config[:project_funders],
       )
       applicability.call(opportunity)
     end
+
+    def self.for_opportunity(opportunity)
+      all_rules = Hmis::Ce::Match::Rule.preload(:owner).order(:owner_type, :id).to_a
```

Comment:
adjust to allow class method to apply to a scope. Otherwise the behavior might be non-intuitive (should support invocation like `Hmis::Ce::Match::Rule.where(owner: owner).for_opportunity(opportunity)`)
```suggestion
      all_rules = preload(:owner).order(:owner_type, :id).to_a
```

Reply:
I think the convention would also be to make this a `scope`

Reply:
My general understanding is that if you're returning an ActiveRecord::Relation, then a scope is most correct.

If you're returning something else that can't be chained (such as an array in this case), a class method is the more appropriate choice.

Of course either will work, I don't think there's much of a functional difference; it's more about conveying intent.

So if changing this to a scope, I'd suggest returning a scope of Opportunity.where(id:...) rather than an array

[rails docs](https://api.rubyonrails.org/classes/ActiveRecord/Scoping/Named/ClassMethods.html#method-i-scope)
> scope() 
Adds a class method for retrieving and querying objects. The method is intended to return an [ActiveRecord::Relation](https://api.rubyonrails.org/classes/ActiveRecord/Relation.html) object, which is composable with other scopes

Reply:
oh I see, thanks. seems like it can remain a class method then.